client/src/components/entities/Description.tsx

```diff
@@ -16,105 +16,122 @@
-    WebkitBoxOrient: "vertical", // eslint-disable-line
-    minHeight: isHeightFixed ? `${25 * numberLines}px` : undefined,
-    height: isHeightFixed ? `${25 * numberLines}px` : undefined,
-  };
+  let content: React.ReactNode = null;
 
-  const markdownDescription =
-    description && typeof description === "string" ? (
-      <Fragment>
+  if (description && typeof description === "string") {
+    content = (
+      <>
         <RenkuMarkdown
           markdownText={description}
           singleLine={numberLines === 1}
-          style={descriptionStyles}
         />
         <span className="ms-1">
           {description.includes("\n") ? " [...]" : ""}
         </span>
-      </Fragment>
-    ) : (
-      description
+      </>
     );
-
-  const isUpdatingValue = useSelector(
-    (state: RootStateOrAny) =>
-      state.stateModel?.project?.metadata?.description?.updating
-  );
-  if (isUpdatingValue) {
-    return (
-      <div
-        className="card-text text-rk-text-light"
-        style={descriptionStyles}
-        data-cy="updating-description"
-      >
-        <small>
-          <i>Updating description...</i>
-        </small>
-      </div>
+  } else if (description) {
+    content = <span>{description}</span>;
+  } else if (loading) {
+    content = (
+      <small className="card-text text-rk-text-light">
+        <i>Loading description...</i>
+      </small>
+    );
+  } else if (unavailable) {
+    content = (
+      <small className="card-text text-rk-text-light">
+        <i>(Description unavailable: {unavailable})</i>
+      </small>
+    );
+  } else if (showSuggestion && hasDevAccess && urlChangeDescription) {
+    content = (
+      <i>
+        (This project has no description. You can provide one{" "}
+        <Link to={urlChangeDescription}>here</Link>.)
+      </i>
     );
   }
 
+  return (
+    <EntityDescriptionContainer
+      className={className}
+      isHeightFixed={isHeightFixed}
+      numberLines={numberLines}
+    >
+      {content}
+    </EntityDescriptionContainer>
+  );
+}
+
+interface EntityDescriptionContainerProps {
+  children: React.ReactNode;
+  className?: string;
+  isHeightFixed: boolean;
+  numberLines: number;
+}
+
+function EntityDescriptionContainer({
+  children,
+  className,
+  isHeightFixed,
+  numberLines,
+}: EntityDescriptionContainerProps) {
+  const descriptionStyles: CSSProperties = {
+    display: "-webkit-box",
+    height: isHeightFixed ? `${25 * numberLines}px` : undefined,
+    lineClamp: isHeightFixed ? numberLines : undefined,
+    margin: "12px 0 0 0",
+    minHeight: isHeightFixed ? `${25 * numberLines}px` : undefined,
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    WebkitBoxOrient: "vertical",
+    WebkitLineClamp: isHeightFixed ? numberLines : undefined,
+  };
+
   return (
     <div
-      className={`card-text ${className}`}
+      className={cx("card-text", className)}
       style={{ ...descriptionStyles, margin: "12px 0 0 0" }}
```

Comment:
Margin is added again here. We can just have:

```suggestion
      style={style}
```